Source/FullScreenPokemon.ts

```diff
@@ -4888,6 +4888,55 @@ module FullScreenPokemon {
             player.FSP.killNormal(player.bordering[player.direction]);
         }
 
+        /**
+         * Makes a StrengthBoulder move.
+         *
+         * @param player   The Player.
+         * @param pokemon   The Pokemon using Strength.
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         * @todo Add an animation for what happens when the StrengthBoulder is moved.
+         */
+        partyActivateStrength(player: IPlayer, pokemon: IPokemon): void {
+            player.FSP.MenuGrapher.registerB();
+            player.FSP.MenuGrapher.registerB();
+            player.FSP.closePauseMenu();
+            player.FSP.TimeHandler.addEventInterval(
+                function (): void {
+                    var boulder: IHMSolid = <IHMSolid>player.bordering[player.direction];
+
+                    switch (player.direction) {
+                        case 0:
+                            boulder.xvel = 0;
+                            boulder.yvel = 8 * -boulder.FSP.unitsize;
+                            boulder.FSP.shiftBoth(boulder, boulder.xvel, boulder.yvel);
+                            break;
+
+                        case 1:
+                            boulder.xvel = 8 * boulder.FSP.unitsize;
+                            boulder.yvel = 0;
+                            boulder.FSP.shiftBoth(boulder, boulder.xvel, boulder.yvel);
+                            break;
+
+                        case 2:
+                            boulder.xvel = 0;
+                            boulder.yvel = 8 * boulder.FSP.unitsize;
+                            boulder.FSP.shiftBoth(boulder, boulder.xvel, boulder.yvel);
+                            break;
+
+                        case 3:
+                            boulder.xvel = 8 * -boulder.FSP.unitsize;
+                            boulder.yvel = 0;
+                            boulder.FSP.shiftBoth(boulder, boulder.xvel, boulder.yvel);
+                            break;
+
+                        default:
+                            throw new Error("Unknown direction: " + player.direction + ".");
+                    }
+                },
+                1,
+                0);
```

Comment:
Instead of doing one movement of 8 * unitsize, it should be doing multiple movements of some fraction of that (probably eight of 1?).